hedera-node/hedera-app/src/xtest/java/contract/Erc721OperationsTest.java

```diff
@@ -255,51 +200,70 @@ private TransactionBody synthSetApprovalForAll(
     private ContractCallTransactionBody callWithParams(@NonNull final ByteBuffer encoded) {
         return ContractCallTransactionBody.newBuilder()
                 .functionParameters(Bytes.wrap(encoded.array()))
-                .contractID(ERC721_FULL_CONTRACT)
-                .gas(GAS_TO_OFFER)
+                .contractID(Erc721OperationsConstants.ERC721_FULL_CONTRACT)
+                .gas(Erc721OperationsConstants.GAS_TO_OFFER)
                 .build();
     }
 
-    private void assertContractDeployedWithExpectedStorage() throws IOException {
+    private void assertExpectedContract() {
         // Assert the contract exists in state at the expected id number
-        final ReadableKVState<AccountID, Account> accounts = scaffoldingComponent
-                .hederaState()
-                .createReadableStates(TokenServiceImpl.NAME)
-                .get(TokenServiceImpl.ACCOUNTS_KEY);
-        final var contract = accounts.get(ERC721_FULL);
+        final var contract = expectedDeployedContract();
         assertNotNull(contract);
         assertTrue(contract.smartContract());
-        assertEquals(contract.contractKvPairsNumber(), EXPECTED_STORAGE.size());
+    }
 
-        // Assert its deployed bytecode matches expectations
+    private void assertExpectedBytecode() throws IOException {
         final ReadableKVState<EntityNumber, Bytecode> bytecode = scaffoldingComponent
                 .hederaState()
                 .createReadableStates(ContractServiceImpl.NAME)
                 .get(ContractSchema.BYTECODE_KEY);
-        final var actualBytecode = bytecode.get(new EntityNumber(ERC721_FULL.accountNumOrThrow()));
+        final var actualBytecode =
+                bytecode.get(new EntityNumber(Erc721OperationsConstants.ERC721_FULL.accountNumOrThrow()));
         assertNotNull(actualBytecode);
         assertEquals(expectedBytecode(), actualBytecode.code());
+    }
 
-        // Assert its storage matches expectations
+    private void assertExpectedStorage() {
         final ReadableKVState<SlotKey, SlotValue> storage = scaffoldingComponent
                 .hederaState()
                 .createReadableStates(ContractServiceImpl.NAME)
                 .get(ContractSchema.STORAGE_KEY);
-        EXPECTED_STORAGE.forEach((key, value) -> {
-            final var slot = storage.get(new SlotKey(ERC721_FULL.accountNumOrThrow(), key));
+        Erc721OperationsConstants.EXPECTED_STORAGE.forEach((key, value) -> {
```

Comment:
What if the actual contract store has _extra_ slot/values?

(There _was_ a test for expected K/V pairs# above previously ...)

(This method not currently called but I assume it will be shortly ...)

Reply:
Correct; this assertion is added [here](https://github.com/hashgraph/hedera-services/blob/07825-xtest-calls-and-storage/hedera-node/hedera-app/src/xtest/java/contract/Erc721OperationsTest.java#L234) in follow-up [PR](https://github.com/hashgraph/hedera-services/pull/7854).